sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -310,6 +311,19 @@ def encode(self, value):
     return value.SerializePartialToString(deterministic=True)
 
 
+class ProtoPlusCoderImpl(SimpleCoderImpl):
+  """For internal use only; no backwards-compatibility guarantees."""
+  def __init__(self, proto_plus_type):
+    # type: (Type[proto.Message]) -> None
+    self.proto_plus_type = proto_plus_type
+
+  def encode(self, value):
+    return value._pb.SerializePartialToString(deterministic=True)
+
+  def decode(self, value):
+    return self.proto_plus_type.deserialize(value)
```

Comment:
To confirm, the API to be used here is different from ProtoCoder above (which uses 'ParseFromString') ?

Reply:
This is the preferred way to deserialize proto plus types. I can't use the serialize method on line 321 because it is not deterministic, so must drop down to the protobuf interface to do so.